certbot/eff.py

```diff
@@ -71,11 +71,17 @@ def _check_response(response):
 
     """
     logger.debug('Received response:\n%s', response.content)
-    if response.ok:
-        if not response.json()['status']:
-            _report_failure('your e-mail address appears to be invalid')
-    else:
+    try:
+        response.raise_for_status()
+        assert response.json()['status'] == True
+    except requests.exceptions.HTTPError:
         _report_failure()
+    except ValueError:
+        _report_failure('there was a problem with the server response')
+    except KeyError:
+        _report_failure('the server response was missing some information')
```

Comment:
I'd personally handle `ValueError` and `KeyError` in the same way and say there was a problem with the server response. I personally think this is a cleaner, user facing error message because exactly what the problem was isn't something the user can do anything about.

I don't have a strong preference though so if you like things better this way, it's fine with me.